dockertest/environment.py

```diff
@@ -210,28 +210,42 @@ def __new__(cls, *args, **dargs):
         else:
             _object = ''  # default
 
-        _clues = cls.investigate(str(_object))
-        victims = set(_clues.values())
-        if len(victims) > 1:
-            raise ValueError("Multiple distros detected: %s" % _clues)
-        elif len(victims) != 1:
-            raise ValueError("No distros detected: %s" % _clues)
+        _matches = cls.investigate(str(_object))
+        distro_matches = set(_matches.values())
+        if len(distro_matches) > 1:
+            raise ValueError("Multiple distros detected: %s" % _matches)
+        elif len(distro_matches) != 1:
+            raise ValueError("No distros detected: %s" % _matches)
 
         # immutables can't rely on __init__(), it must be done here
-        new_cd = super(CanonicalDistro, cls).__new__(cls, victims.pop())
-        new_cd.clues = _clues
+        new_cd = super(CanonicalDistro, cls).__new__(cls, distro_matches.pop())
+        new_cd.matches = _matches
         return new_cd
 
+    # Must exactly match the distros list defined above: one method
+    # for each distro. FIXME: define what "hints" is supposed to be.
```

Comment:
It's defined (minus the name) in the class-docstring:

>      #: Each must accept a single, optional string-argument for
>      #: testing instead of whatever default data would normally
>      #: be used.

Hmmm, maybe "instead" is too strong a word.  Maybe: "...for examining prior to default metadata."